app/models/shipit/undeployed_commit.rb

```diff
@@ -11,15 +11,14 @@ def deploy_state(bypass_safeties = false)
       state = deployable? ? 'allowed' : status.state
       unless bypass_safeties
         state = 'deploying' if stack.active_task?
-        state = 'locked' if stack.locked?
+        state = 'locked' if locked?
```

Comment:
is the `stack.locked?` not important anymore? maybe `locked? || stack.locked?` would be better?